src/metabase/driver/sql_jdbc/execute.clj

```diff
@@ -71,13 +71,38 @@
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
+(defmulti ^Statement statement-supported?
+  "Indicates whether the given driver supports creating a java.sql.Statement, via the Connection. By default, this is
+  true for all :sql-jdbc drivers.  If the underlying driver does not support Statement creation, override this as
+  false."
+  {:added "0.39.0", :arglists '([driver])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmulti ^Statement statement
+  "Create a Statement object using the given connection. Only called if statement-supported? above returns true. This
+  is to be used to execute native queries, which implies there are no parameters. As with prepared-statement, you
+  shouldn't need to override the default implementation for this method; if you do, take care to set options to maximize
+  result set read performance (e.g. `ResultSet/TYPE_FORWARD_ONLY`); refer to the default implementation."
+  {:added "0.39.0", :arglists '(^java.sql.Statement [driver ^java.sql.Connection connection])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
 (defmulti execute-query!
   "Execute a `PreparedStatement`, returning a `ResultSet`. Default implementation simply calls `.executeQuery()`. It is
   unlikely you will need to override this."
   {:added "0.35.0", :arglists '(^java.sql.ResultSet [driver ^java.sql.PreparedStatement stmt])}
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
+(defmulti execute-select!
```

Comment:
I think this is a little confusing in combination with `execute-query!`, since they're both executing `SELECT` queries. Maybe `execute-statement!` and `execute-prepared-statement!`?